### DISCUSSION TITLE
Custom component and utilitity classes generated in wrong order

### DISCUSSION THREAD

**Discussion by zipper** [tailwindlabs/tailwindcss#10456] [Help]
Hello.

I have a custom component, where I convert table to grid display. When I add the typography plugin to Tailwind configuration, the order of generated CSS is wrong when responsive variants are used. The component is _after_ the utilities. This happens only when I use the responsive variants, like `md:`. More over, when I tried to simulate this on Tailwind playground, I failed. Everything works fine there, as you can see here https://play.tailwindcss.com/VFtNNQGBfl.

But when I use this in my local build, the padding utility is overridden because of the order. I'm using webpack, I removed everything I imagined could interfere (eg. Sass, custom variants, ...), but still getting the generated order wrong. I reduced my `tailwind.config.js` to this (pretty much the same as on playground):

```js
/** @type {import('tailwindcss').Config} */
module.exports = {
	content: [
		"test.html"
	],
	plugins: [
		require('@tailwindcss/typography'),
	],
}
```

The test.html contains same code as playground demo. When I comment out the require for typography, order is correct and padding from utility is applied. I would be thankful for any help and suggestion, where the problem might be. When I move the code into base layer, it works even with typography, but I would rather have it inside components layer. Thanks in advance!

**Comment by wongjn** [comment]
Please consider providing a remotely-hosted git repository that reproduces the unexpected behavior.

**Reply by zipper** [reply]
I'll try to as soon as will have some time 👍

**Comment by zipper** [comment]
I have finally had some time to create a gist. https://gist.github.com/zipper/b351a3a278c2002dcd3e0ff6c4ba0426. To reproduce a bug:
1. Download gist
2. Run `npm install && npm run start`. That will create `dist` folder with CSS file.
3. Open `index.html` in a browser.

There is a table, which is styled to use a grid using custom component (with variant) `sm:table-grid` (in real project, I use this on small screens only to wrap the table) and then there are `sm:p-8` classes on rows. But those are overridden by default styles from component `table-grid`. Component is correctly "registered" inside `@layer components`, so this should not happen.

Why I think this is a bug:
1. Comment out the `require('@tailwindcss/typography'),` in `tailwind.config.js:7` and rebuild the styles. Now the utility padding is properly after the custom component styles in dist CSS.
2. Or change the `@layer components` to `@layer base`, rebuild styles and the utility padding is again properly after custom component styles.
